Add doc dir for Keptn 0.16 (#1228)
keptn/keptn.github.io: content/docs/0.16.x/manage/service/index.md

---
title: Service
description: Create a service in Keptn.
weight: 30
keywords: [0.16.x-manage]
aliases:
---

After creating a project, Keptn allows creating a service into Keptn. 

## Service name restrictions

* Service name must be a valid unix directory name (*Note*: for each service a directory with the corresponding name is created in the upstream git repository)
* Keptn Version >= 0.8.3:
  Service name must be less than 43 characters (*Note*: template was reduced to `${SERVICE}-generated` to allow longer service names)

## Create a service

* To create a service, use the [keptn create service](../../reference/cli/commands/keptn_create_service) command and provide the service and project name (`--project` flag): 

```console
keptn create service SERVICENAME --project=PROJECTNAME
```

**Requirements for the Helm Chart to deploy**

> After creating a service, you need to provide a Helm Chart for the service to deploy it. For Keptn, the [Helm Chart](https://Helm.sh/) has the following requirements:

Reviewer: ```suggestion
After creating a service, you need to provide a Helm Chart where the service deploys. For Keptn, the [Helm Chart](https://Helm.sh/) has the following requirements:
```

> 1. The Helm chart _has_ to contain exactly one [deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/). In this deployment, the properties `spec.selector.matchLabels.app` and `spec.template.metadata.labels.app` have to be set.

Reviewer: ```suggestion
1. The Helm chart _must_ contain exactly one [deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/). In this deployment, the properties `spec.selector.matchLabels.app` and `spec.template.metadata.labels.app` must be set.
```

> 1. The Helm chart _has_ to contain exactly one [service](https://kubernetes.io/docs/concepts/services-networking/service/). In this service, the property `spec.selector.app` has to be set.

Reviewer: ```suggestion
1. The Helm chart _must_ contain exactly one [service](https://kubernetes.io/docs/concepts/services-networking/service/). The property `spec.selector.app` must be set in this service.
```

> 1. The Helm chart _has_ to contain a `values.yaml` file with at least the `image` and `replicaCount` parameter for the deployment. These `image` and `replicaCount` parameters have to be used in the deployment. An example is shown below:

Reviewer: ```suggestion
1. The Helm chart _must_ contain a `values.yaml` file with at least the `image` and `replicaCount` parameter wfor the deployment. These `image` and `replicaCount` parameters must be used in the deployment. An example is shown below:
```
